src/util.ts

```diff
@@ -708,3 +709,45 @@ export function getMlPoweredJsQueriesStatus(config: Config): string {
       return "other";
   }
 }
+
+/**
+ * Prompt the customer to upgrade to CodeQL Action v2, if appropriate.
+ *
+ * Check whether a customer is running v1. If they are, and we can determine that the GitHub
+ * instance supports v2, then log a warning about v1's upcoming deprecation prompting the customer
+ * to upgrade to v2.
+ */
+export async function checkActionVersion(version: string) {
+  if (!semver.satisfies(version, ">=2")) {
+    const githubVersion = await api.getGitHubVersionActionsOnly();
+    // Only log a warning for versions of GHES that are compatible with CodeQL Action version 2.
+    if (
+      githubVersion.type === GitHubVariant.DOTCOM ||
+      githubVersion.type === GitHubVariant.GHAE ||
+      (githubVersion.type === GitHubVariant.GHES &&
+        semver.satisfies(
+          semver.coerce(githubVersion.version) ?? "0.0.0",
+          ">=3.4"
+        ))
+    ) {
+      core.warning(
+        "CodeQL Action version 1 will be deprecated on December 7th, 2022. Please upgrade to " +
+          "version 2. For more information, see " +
+          "https://github.blog/changelog/2022-04-27-code-scanning-deprecation-of-codeql-action-v1/."
```

Comment:
Nitpicking, but easier to copy-paste.
```suggestion
          "https://github.blog/changelog/2022-04-27-code-scanning-deprecation-of-codeql-action-v1/"
```